internal/component_options.go

```diff
@@ -77,6 +77,11 @@ type DeleteFileOptions struct {
 	Name string
 }
 
+type DownloadFileOptions struct {
+	Name   string
+	Handle *handlemap.Handle
+}
+
```

Comment:
Since the name is already in `handle.Path`, you could drop this struct altogether.
This comment is the other half of the comment I left on `DownloadFile` above.